src/app/data/data.service.ts

```diff
@@ -22,25 +23,97 @@ export class DataService {
   activeBubbleHighlight: MapDataAttribute = BubbleAttributes[0];
   mapView;
-  mapConfig = {
-    style: './assets/style.json',
-    center: [-98.5556199, 39.8097343],
-    zoom: 3,
-    minZoom: 3,
-    maxZoom: 14
-  };
+  mapConfig;
   private mercator = new SphericalMercator({ size: 256 });
   private tileBase = 'https://s3.us-east-2.amazonaws.com/eviction-lab-tilesets/fixtures/';
   private tilePrefix = 'evictions-';
   private tilesetYears = ['90', '00', '10'];
-  private maxZoom = 10;
-  constructor(private http: Http) { }
+  private queryZoom = 10;
+
+  constructor(private http: Http) {}
+
+  /**
+   * Sets the choropleth layer based on the provided `DataAttributes` ID
+   * @param id string corresponding to the `MapDataAttribute` in `DataAttributes`
+   */
+  setChoroplethHighlight(id: string) {
+    const dataAttr = this.dataAttributes.find((attr) => attr.id === id);
+    if (dataAttr) {
+      this.activeDataHighlight = dataAttr;
+    }
+  }
+
+  /**
+   * Sets the bubble layer based on the provided `BubbleAttributes` ID
+   * @param id string corresponding to the `MapDataAttribute` in `BubbleAttributes`
+   */
+  setBubbleHighlight(id: string) {
+    const bubbleAttr = this.bubbleAttributes.find((attr) => attr.id === id);
+    if (bubbleAttr) {
+      this.activeBubbleHighlight = bubbleAttr;
+    }
+  }
+
+  /**
+   * Sets the layer geography based on the provided `DataLevels` ID
+   * @param id string of the MapLayerGroup in `DataLevels`
+   */
+  setGeographyLevel(id: string) {
+    const geoLevel = this.dataLevels.find((level) => level.id === id);
+    if (geoLevel) {
+      this.activeDataLevel = geoLevel;
+    }
+  }
+
+  /** */
+  setLocations(locations) {
+    locations.forEach(l => {
+      this.getTileData(l.layer, l.lonLat, true)
+        .subscribe((data) => { this.addLocation(data); });
+    });
+  }
+
+  /**
+   * Sets the bounding box for the map to focus to
+   * @param mapBounds an array with four coordinates representing south, west, north, east
+   */
+  setMapBounds(mapBounds) {
+    this.mapView = mapBounds;
+  }
+
+  /**
+   * Returns the URL parameters for the current view
+   */
+  getUrlParameters() {
+    const paramMap = [ 'locations', 'year', 'geography', 'type', 'choropleth', 'bounds' ];
+    return this.getRouteArray().reduce((a, b, i) => {
+      return a + ';' + paramMap[i] + '=' + b;
+    }, '');
+  }
+
+  /**
+   * Gets an array of values that represent the current route
+   */
+  getRouteArray() {
+    const locations = this.activeFeatures.map((f, i, arr) => {
+      const xy = this.getFeatureXY(f);
+      return f.properties['layerId'] + ',' + xy.x + ',' + xy.y;
+    }).join('+');
+    return [
+      (locations === '' ? 'none' : locations),
+      this.activeYear,
+      this.activeDataLevel.id,
+      this.stripYearFromAttr(this.activeBubbleHighlight.id),
+      this.stripYearFromAttr(this.activeDataHighlight.id),
+      this.mapView ? this.mapView.join() : null
```

Comment:
Not sure if it'll cause the map to jump around, but could we simplify the coordinates here to 2 or 3 decimal places? The URL looks a lot more unwieldy as is

Reply:
Yeah, makes sense to me.  Will update.